python/tank/deploy/tank_commands/unregister_folders.py

```diff
@@ -0,0 +1,311 @@
+# Copyright (c) 2013 Shotgun Software Inc.
+# 
+# CONFIDENTIAL AND PROPRIETARY
+# 
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit 
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your 
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights 
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+"""
+Method to unregister folders from the path cache
+"""
+
+from ...errors import TankError
+from ... import path_cache
+from .action_base import Action
+from ...util.login import get_current_user 
+
+class UnregisterFoldersAction(Action):
+    """
+    Tank command for unregistering a folder on disk from Shotgun. This is part of the process of
+    the deletion of a folder on disk. As part of removing or moving, the folder needs to be 
+    unregistered with Shotgun to ensure that the connection between that path and the related 
+    entity is undone.
+    """
+    
+    def __init__(self):
+        """
+        Constructor
+        """        
+        Action.__init__(self, 
+                        "unregister_folders", 
+                        Action.TK_INSTANCE, 
+                        ("Unregisters the folders for an object in Shotgun."), 
+                        "Admin")
+        
+        # this method can be executed via the API
+        self.supports_api = True
+
+        self.parameters = {}
+        
+        self.parameters["path"] = { "description": "Path to unregister. Any child paths will be unregistered too.",
+                                    "default": None,
+                                    "type": "str" }
+        
+        self.parameters["entity"] = { "description": ("Entity to unregister. Should be a Shotgun style entity "
+                                                      "dictionary with keys type and id."),
+                                      "default": None,
+                                      "type": "dict" }
+        
+    
+    def run_interactive(self, log, args):
+        """
+        Tank command accessor
+        """
+        
+        if not self.tk.pipeline_configuration.get_shotgun_path_cache_enabled():            
+            # remote cache not turned on for this project
+            log.error("Looks like this project doesn't synchronize its folders with Shotgun! "
+                      "If you want to turn on synchronization for this project, run "
+                      "the 'upgrade_folders' tank command.")
```

Comment:
This error implies they're completely missing path cache functionality. It might be nicer to inform them this is new, upgraded functionality. 

> Looks like this project doesn't synchronize its folders with Shotgun! If you'd like to upgrade your path cache to turn on synchronization for this project...

Also, the `tank` command examples we use in most places are worded like so:

> run the `tank upgrade_folders` command.